libraries/AP_AHRS/AP_AHRS_NavEKF.cpp

```diff
@@ -1974,6 +1988,22 @@ bool AP_AHRS_NavEKF::get_hgt_ctrl_limit(float& limit) const
 // this is not related to terrain following
 void AP_AHRS_NavEKF::set_terrain_hgt_stable(bool stable)
 {
+    switch (terrainHgtStableState) {
```

Comment:
Needs a comment.
```suggestion
    // avoid repeatedly setting variable in NavEKF objects to prevent spurious event logging
    switch (terrainHgtStableState) {
```

Reply:
done